File: app/src/main/java/com/mifos/apache/fineract/ui/online/loans/loanapplication/OnBottomSheetDialogListener.java
package com.mifos.apache.fineract.ui.online.loans.loanapplication;

import com.mifos.apache.fineract.data.models.loan.CreditWorthinessFactor;

/**
 * @author Rajan Maurya
 *         On 23/07/17.
 */
public interface OnBottomSheetDialogListener {

    interface AddDebt {
        void addDebt(CreditWorthinessFactor creditWorthinessFactor);
        void editDebt(CreditWorthinessFactor creditWorthinessFactor, int position);
    }

    interface AddIncome {
        void addIncome(CreditWorthinessFactor creditWorthinessFactor);
        void editIncome(CreditWorthinessFactor creditWorthinessFactor, int position);
    }
}

File: app/src/main/java/org/apache/fineract/ui/online/customers/customerpayroll/editcustomerpayroll/editpayrollbottomsheet/OnBottomSheetDialogListener.kt
package org.apache.fineract.ui.online.customers.customerpayroll.editcustomerpayroll.editpayrollbottomsheet

import org.apache.fineract.data.models.payroll.PayrollAllocation


interface OnBottomSheetDialogListener {

    fun editPayrollAllocation(payrollAllocation: PayrollAllocation, position: Int)

    fun addPayrollAllocation(payrollAllocation: PayrollAllocation)
}
